Source/WebCore/platform/graphics/cg/GraphicsContextCG.cpp

```diff
@@ -211,6 +211,23 @@ CGContextRef GraphicsContextCG::platformContext() const
     return m_data->m_cgContext.get();
 }
 
+const DestinationColorSpace& GraphicsContextCG::colorSpace() const
+{
+    if (m_colorSpace)
+        return *m_colorSpace;
+
+    auto context = platformContext();
+
+    if (CGContextGetType(context) == kCGContextTypeIOSurface)
```

Comment:
What about PDF contexts? Would it be better to pass in the colorspace at the point where we wrap a CGContextRef in a GraphicsContext, where we (usually) have a better idea of what kind of context it is?

Reply:
Can not `CGContextCopyDeviceColorSpace() `be used for getting the color space of PDF contexts? I also do not know how to create a PDF context to test it. And I do not see that we create one in WebKit.

Reply:
Print to PDF, and we have API for it.

Reply:
I added a FIXME comment for the PDF color space to be investigated later.